palette/src/gradient/mod.rs

```diff
@@ -12,6 +13,61 @@ use crate::float::Float;
 use crate::Mix;
 use crate::{from_f64, FromF64};
 
+#[cfg(feature = "named_gradients")]
+pub mod named;
+
+/// Types which represent an ordered collection of known size
+pub trait ArrayLike : Index<usize>{
```

Comment:
After looking at this again, it seems to me like we could replace it with `AsRef<[T]>` or `AsMut<[T]>`. That would remove the const generics dependency and lower the required Rust version.

Reply:
I thought about using `AsRef<[T]>`  instead of creating a custom trait. You already mentioned the advantages. The disadvantages would be to have to call `as_ref()` everytime we would want to use the collections, we can't know if `Gradient` would own it's collection and it would be less generic, because the underlying data would need to act like a slice (so generators would be difficult to implement). These are all minor, so to be honest I'm not sure why I decided against it.

I will try to replace the custom trait with `AsRef<[T]>` in the next few days and comment here again afterwards.

Reply:
It would be great if you can give it a try! 👍 Calling `as_ref` is probably not a big deal compared to maintaining a custom trait, to be honest, so I prefer `AsRef<T>` if it works. It's usually only necessary to do it once per function, though.

You are right that generators will be excluded in this case, but a workaround would be to collect values in an array or `Vec` and pass that in. As a slice, even, to save some allocations.

Reply:
The latest commit removes the custom trait in favor of `AsRef<[T]>`. To make it work, I added `Phantomdata<Color>` to the Gradient. Other than that, all changes were straightforward.